src/components/TabPaneTabContextMenu.tsx

```diff
@@ -0,0 +1,85 @@
+import { Item, ItemParams, Menu, PredicateParams, Separator } from 'react-contexify';
+import { VscClose, VscCloseAll, VscLayoutSidebarLeft, VscLayoutSidebarRight } from 'react-icons/vsc';
+
+import { useOpenEditorsDataForPane } from '../atoms/hooks/useOpenEditorsDataForPane';
+import { EditorId } from '../atoms/types/EditorData';
+import { PaneId } from '../atoms/types/PaneGroup';
+import { emitEvent } from '../events';
+
+export const TABPANE_TAB_MENU_ID = 'TabPaneContextMenu';
+
+export interface TabPaneTabContextMenuProps {
+  editorId: EditorId;
+  paneId: PaneId;
+}
+
+export function TabPaneTabContextMenu() {
+  const leftOpenEditorsData = useOpenEditorsDataForPane('LEFT');
+
+  const canMoveEditorToPane = (targetPane: PaneId) => (params: PredicateParams<TabPaneTabContextMenuProps>) => {
+    const isInLeft = leftOpenEditorsData.some((e) => e.id === params.props?.editorId);
+    if ((isInLeft && targetPane === 'RIGHT') || (!isInLeft && targetPane === 'LEFT')) {
+      return false;
+    }
+    return true;
```

Comment:
I think the function returns whether the editor **cannot** move to the given pane instead of whether it can.
You could simply do `return (isInLeft && targetPane === 'RIGHT') || (!isInLeft && targetPane === 'LEFT')`